crates/harp/src/error.rs

```diff
@@ -266,3 +268,15 @@ impl From<Utf8Error> for Error {
         Self::InvalidUtf8(error)
     }
 }
+
+fn truncate_lines(text: String, max_lines: usize) -> String {
+    let n_lines = text.lines().count();
+    if n_lines <= max_lines {
+        return text;
+    }
+
+    let mut text: String = text.lines().take(max_lines).collect();
+    text.push_str(&format!("... *Truncated {} lines*", n_lines - max_lines));
```

Comment:
Should there be a leading `\n` to put this on its own line?

Reply:
yep!